bindings/sundials4py/nvector/nvector_manyvector.cpp

```diff
@@ -0,0 +1,31 @@
+#include "sundials4py.hpp"
+
+#include <nvector/nvector_manyvector.h>
+#include <sundials/sundials_core.h>
+#include <sundials/sundials_nvector.hpp>
+
+namespace nb = nanobind;
+using namespace sundials::experimental;
+
+namespace sundials4py {
+
+void bind_nvector_manyvector(nb::module_& m)
+{
+#include "nvector_manyvector_generated.hpp"
+
+  m.def(
+    "N_VNew_ManyVector",
+    [](sunindextype num_subvectors, std::vector<N_Vector> vec_array_1d,
+       SUNContext sunctx) -> std::shared_ptr<std::remove_pointer_t<N_Vector>>
+    {
+      N_Vector* vec_array_1d_ptr = reinterpret_cast<N_Vector*>(
+        vec_array_1d.empty() ? nullptr : vec_array_1d.data());
+      return our_make_shared<std::remove_pointer_t<N_Vector>, N_VectorDeleter>(
+        N_VNew_ManyVector(num_subvectors, vec_array_1d_ptr, sunctx));
+    },
+    nb::arg("num_subvectors"), nb::arg("vec_array_1d"), nb::arg("sunctx"),
+    nb::keep_alive<0, 3>() /* keep the SUNContext alive as long as the N_Vector is */,
+    nb::keep_alive<0, 2>() /* keep the list, and thus the elements, alive as long as the N_Vector is */);
```

Comment:
I think these comments are backwards and should be
```suggestion
    nb::keep_alive<0, 3>() /* keep the N_Vector alive as long as the N_Vector SUNContext is alive */,
    nb::keep_alive<0, 2>() /* keep the N_Vector alive as long as the list, and thus the elements, is alive */);
```
Or am I misunderstanding the `keep_alive` syntax?

Reply:
Disregard this suggestion, I was confusing the ordering with the `returns_references_to` syntax which reverses the nurse and patient parameter ordering

Reply:
Yeah, it is a bit confusing the but the syntax has to be reversed because a different argument needs to be the variadic one.